MetaMask/metamask-mobile
# feat(ramp): popular region and search states

## **Description**

This PR improves the Region selector for Ramp flows by removing duplicate code and adding these features:
1. Default list is now shown in sections (Popular Regions and Regions)
2. Searching will also match and display states

## **Related issues**

Fixes: #

## **Manual testing steps**

1. Go to Ramp flow
2. Open a Region modal (Region or Amount screens)
3. Confirm the changes from descriptions

## **Screenshots/Recordings**



### **Before**
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/224e7ecb-9198-41fc-84f2-33254c4b4ca7" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/033bf109-270c-4679-bfab-287733f05219" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/ccef6893-a40a-4a7c-afcd-d3968e307fa7" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/22c3e7c0-7bf3-446e-98c7-2bf43a45be3f" />



### **After**

<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/4dc21a19-dc7e-4154-b19b-4785ad3c8be9" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/0a0ef8dc-2c82-4fe9-994a-277976055297" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/9affd7e1-1edb-48ea-bb63-f902ca0ff26c" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/57da0148-ce19-4d6f-9807-133cdb1331c2" />
<img width="135" src="https://github.com/MetaMask/metamask-mobile/assets/1024246/b666caec-5e5e-4cdb-b4e8-dd3339f59f69" />


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
